### PR TITLE
feat:add ticket merge

### PR DESCRIPTION
**Pull request by welmaris**
### Issue number

DATA-2072

### Description of feature

When running merge_double_tickets, all tickets with the same nerds_ticket_id should be merged. it will fill in the empty values where applicable. Notes should get the association to the ticket that stays and description (content) is added if it's not the same.

### Other info

This method doesn't require anything from Wherescape and is more of a Hubspot functionality. Therefore, I've left the access_token as a required parameter rather than retrieving it from Wherescape.

### CODE REVIEW

File: wherescape/connectors/hubspot/hubspot_wrapper.py [hunk 1]
```diff
@@ -1,51 +1,64 @@
-import hubspot
 import logging
+from enum import StrEnum, auto
+
+import hubspot.crm
+import hubspot.crm.companies
```
Comment (Shrikey) [inline wherescape/connectors/hubspot/hubspot_wrapper.py]: double import?

File: wherescape/connectors/hubspot/hubspot_wrapper.py [hunk 2]
```diff
@@ -1,51 +1,64 @@
...
+    HubspotObjectEnum.DEALS: deals.BatchInputSimplePublicObjectBatchInput,
+    HubspotObjectEnum.TICKETS: tickets.BatchInputSimplePublicObjectBatchInput,
 }
 
 
 class Hubspot:
     def __init__(self, access_token: str):
         self.access_token = access_token
-        self.client = hubspot.Client.create(access_token=access_token)
+        self.client: Client = hubspot.client.Client.create(access_token=access_token)
```
Comment (Shrikey) [inline wherescape/connectors/hubspot/hubspot_wrapper.py]: Already imported Client

File: wherescape/connectors/hubspot/hubspot_wrapper.py [hunk 3]
```diff
@@ -81,6 +114,144 @@ def get_properties(self, object_name: str):
...
+
+        except error_api.ApiException as e:
+            logging.error(e)
+
+        logging.info(f"returning {len(results)} items of Hubspot {hs_object}.")
+        return results
+
+    def merge_tickets(self, ticket_a, ticket_b) -> tuple:
+        """
+        This method merges the properties into tickets into the ticket that is the oldest.
```
Comment (Shrikey) [inline wherescape/connectors/hubspot/hubspot_wrapper.py:157]: into tickets into the ticket

Comment (Shrikey) [reply wherescape/connectors/hubspot/hubspot_wrapper.py:157]: The ticket states that the newest should be kept. Is this a conscious choice? What is the reasoning?

File: wherescape/connectors/hubspot/tickets_merge.py [hunk 4]
```diff
@@ -0,0 +1,103 @@
+import logging
+from datetime import datetime
+
+from wherescape.connectors.hubspot.hubspot_wrapper import Hubspot
+
+# Ticket properties we need on want to make sure we have the info in the final ticket.
```
Comment (Shrikey) [inline wherescape/connectors/hubspot/tickets_merge.py]: *on -> or?

**Review by Shrikey — APPROVED** [review]
Look very professional! Found some minor things.

Also please update readme. Include refactoring in it.

Just to be sure: notes are also moved right?

File: wherescape/connectors/hubspot/hubspot_wrapper.py (hunk 3 above)
Comment (welmaris) [reply wherescape/connectors/hubspot/hubspot_wrapper.py:157]: My reasoning is that I read the ticket wrong :sweat_smile:. It's fixed now

**Pull request merged**